### ISSUE TITLE
Elasticsearch output plugin quietly fails when target host is IPv6

### ISSUE THREAD

**Issue by jtackaberry** [fluent/fluent-bit#2345]
Given the following output:

```
[OUTPUT]
    Name            es
    Match           *
    Host            es.example.com
    Port            443
    Replace_Dots    On
    HTTP_User       logs-foo
    HTTP_Passwd     somepassgoeshere
    Logstash_Format On
    Logstash_Prefix logs-foo
    tls             On
    tls.verify      On
    Trace_Output    On
```

And the following (obviously fictional) entry in `/etc/hosts`:

```
2001:dead:beef::1 es.example.com
```

Fluent Bit logs the notorious "failed to flush chunk" warning with no further error message or diagnostic information.

`strace` reveals the problem is that Fluent Bit is creating an `AF_INET` socket but trying to `connect()` using `AF_INET6`.  Obviously the socket should be created `AF_INET6` for this to work.

Strace excerpt:
```
[pid 27799] socket(AF_INET, SOCK_STREAM, IPPROTO_IP) = 38
[pid 27799] fcntl(38, F_GETFL)          = 0x2 (flags O_RDWR)
[pid 27799] fcntl(38, F_SETFL, O_RDWR|O_NONBLOCK) = 0
[pid 27799] setsockopt(38, SOL_TCP, TCP_NODELAY, [1], 4) = 0
[pid 27799] stat("/etc/resolv.conf", {st_mode=S_IFREG|0644, st_size=715, ...}) = 0
[pid 27799] openat(AT_FDCWD, "/etc/hosts", O_RDONLY|O_CLOEXEC) = 39
[pid 27799] fstat(39, {st_mode=S_IFREG|0644, st_size=667, ...}) = 0
[pid 27799] read(39, "# Your system has configured 'ma"..., 4096) = 667
[pid 27799] read(39, "", 4096)          = 0
[pid 27799] close(39)                   = 0
[pid 27799] connect(38, {sa_family=AF_INET6, sin6_port=htons(443), inet_pton(AF_INET6, "2001:dead:beef::1", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, 28) = -1 EAFNOSUPPORT (Address family not supported by protocol)
[pid 27799] close(38)
```

I'm observing this with Elasticsearch but it's possible this problem exists with other outputs as well.

**Comment by edsiper** [comment]
please add ```ipv6 on``` in your output section

**Comment by jtackaberry** [comment]
Aha, thanks @edsiper , that does indeed work.  I guess it's an undocumented option?  I couldn't find `ipv6` anywhere in the docs.

While this does make this particular example work, it's not quite the behavior I was hoping for.  My `es.example.com` has both functioning A and AAAA records, and the host running Fluent Bit is also dual stack.   The desired behavior is that Fluent Bit would try IPs returned by `getaddrinfo()` successively, which may be mixed v4 and v6 addresses, so that Fluent Bit would be robust in the face of any issues with either v4 or v6 path.

**Comment by edsiper** [comment]
This is already fixed in GIT master, it will be part of v1.7 release on ~Feb 2021.

ref: c40c1e26eae100084d0747854c1c2a632d07a6cc
